protobuf-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/protobuf/fromconnectdata/ConnectDataToProtobufDataConverter.java

```diff
@@ -56,4 +66,36 @@ private void addField(final Message.Builder builder, final Field field, final Ob
 
         dataConverter.toProtobufData(schema, value, fieldDescriptor, builder);
     }
+
+    private void addMapField(final Message.Builder builder, final Field field, final Object value) {
+        final String protobufFieldName = field.name();
+        final Schema schema = field.schema();
+        final Descriptors.Descriptor mapDescriptor = builder.getDescriptorForType().findNestedTypeByName(
```

Comment:
What happens if the underlying schema doesn't contain the type?

Reply:
This method will only be invoked when the schema type is MAP. The nested type should always exist as this is how a map type is generated in the Protobuf schema. 

By any chance if the nested type doesn't exist, it will throw a NPE. We can translate to Kafka connect DataException then.